# Dynamic Audification Test

Testing multiple volcano stations with preset configurations


In [7]:
# Import required modules
from python_code.main import main
from python_code.print_manager import print_manager
import os
import ipywidgets as widgets
from IPython.display import display
import subprocess
from datetime import datetime, timedelta, timezone
import matplotlib.pyplot as plt


In [8]:
# Station Presets
STATION_PRESETS = {
    "spurr_seismic": {
        "network": "AV", 
        "station": "SPCN", 
        "channel": "BHZ", 
        "location": "--",
        "name": "Mt. Spurr (Seismic)", 
        "timezone": "America/Anchorage",
        "data_type": "seismic",
        "native_sr": 50  # Hz
    },
    "kilauea_obl_seismic": {
        "network": "HV", 
        "station": "OBL", 
        "channel": "BHZ", 
        "location": "--",
        "name": "Kīlauea Observatory Bluff (Seismic)", 
        "timezone": "Pacific/Honolulu",
        "data_type": "seismic",
        "native_sr": 50  # Hz (estimated)
    },
    "kilauea_uwe_seismic": {
        "network": "HV", 
        "station": "UWE", 
        "channel": "HHZ", 
        "location": "--",
        "name": "Kīlauea Uwekauhane (Seismic)", 
        "timezone": "Pacific/Honolulu",
        "data_type": "seismic",
        "native_sr": 100  # Hz
    },
    "kilauea_uwe_infrasound": {
        "network": "HV", 
        "station": "UWE", 
        "channel": "HDF", 
        "location": "01",
        "name": "Kīlauea Uwekauhane (Infrasound)", 
        "timezone": "Pacific/Honolulu",
        "data_type": "infrasound",
        "native_sr": 100  # Hz
    }
}

# Select your station preset here - CHANGE THIS TO TEST DIFFERENT STATIONS
selected_preset = "spurr_seismic"  # Options: spurr_seismic, kilauea_obl_seismic, kilauea_uwe_seismic, kilauea_uwe_infrasound

station_config = STATION_PRESETS[selected_preset]

print(f"🌋 Selected Station: {station_config['name']}")
print(f"📡 Network.Station.Channel: {station_config['network']}.{station_config['station']}.{station_config['channel']}")
print(f"🕐 Timezone: {station_config['timezone']}")
print(f"📊 Data Type: {station_config['data_type']}")
print(f"🎵 Native Sampling Rate: {station_config['native_sr']} Hz")


🌋 Selected Station: Mt. Spurr (Seismic)
📡 Network.Station.Channel: AV.SPCN.BHZ
🕐 Timezone: America/Anchorage
📊 Data Type: seismic
🎵 Native Sampling Rate: 50 Hz


In [9]:
# Configuration - How much data do you want to process?
days_of_data = 1  # Number of days to process (just like the original!)
audio_sr = 75000  # Hz - our target audio sampling rate

# Calculate speed-up factor for reference
real_time_seconds = days_of_data * 24 * 3600  # Convert days to seconds
native_sr = station_config['native_sr']

# Calculate how many samples we'll have
total_samples = real_time_seconds * native_sr
playback_time_seconds = total_samples / audio_sr
speedup_factor = real_time_seconds / playback_time_seconds

print(f"\n⚡ Audification Analysis:")
print(f"   Real time: {days_of_data} day(s) = {real_time_seconds:,} seconds")
print(f"   Total samples: {total_samples:,} (at {native_sr} Hz)")
print(f"   Playback time: {playback_time_seconds:.1f} seconds (at {audio_sr:,} Hz)")
print(f"   Speed-up factor: {speedup_factor:.0f}x faster")

# Configuration parameters (for future use when main() accepts station params)
config = {
    "days": days_of_data,  # Number of days to process
    "sampling_rate": audio_sr,
    "tick_interval_hours": 1,
    "marker_interval_hours": 1,
    "use_am_pm": True,
    "markers_in_AKST": True,  # We'll need to make this dynamic based on timezone
    "interpolate_missing_data": False
}

print(f"\n🔧 Configuration:")
for key, value in config.items():
    print(f"   {key}: {value}")



⚡ Audification Analysis:
   Real time: 1 day(s) = 86,400 seconds
   Total samples: 4,320,000 (at 50 Hz)
   Playback time: 57.6 seconds (at 75,000 Hz)
   Speed-up factor: 1500x faster

🔧 Configuration:
   days: 1
   sampling_rate: 75000
   tick_interval_hours: 1
   marker_interval_hours: 1
   use_am_pm: True
   markers_in_AKST: True
   interpolate_missing_data: False


In [ ]:
# Configure print options
print_manager.show_times = True          # Time-related messages
print_manager.show_files = True         # File operation messages
print_manager.show_buttons = True        # UI buttons
print_manager.show_marker_summary = True # Marker summaries
print_manager.show_plots = False         # Plot information
print_manager.show_data_info = True      # Data details
print_manager.show_api_requests = True   # API details
print_manager.show_all_markers = False   # All markers detail
print_manager.show_status = True         # Important status messages

print("✅ Print manager configured")


In [ ]:
# Run the audification
print(f"\n🚀 Starting audification for {station_config['name']}...\n")

print(f"⚠️  NOTE: Currently using hardcoded Mt. Spurr settings in main() function")
print(f"   To test other stations, we need to modify main() to accept station parameters\n")

try:
    result = main(**config)
    
    if result and result.get('success'):
        print(f"\n✅ Audification completed successfully!")
        print(f"📁 Files created:")
        print(f"   Audio: {result['audio_file']}")
        print(f"   Plot: {result['plot_file']}")
        print(f"   Markers: {result['marker_file']}")
        print(f"   Data: {result['mseed_file']}")
    else:
        print(f"❌ Audification failed")
        
except Exception as e:
    print(f"❌ Error during audification: {e}")
    result = None


## Notes

### To test different stations:
1. Change the `selected_preset` variable in the second cell
2. Re-run all cells

### Available presets:
- `spurr_seismic` - Mt. Spurr seismic data (50 Hz)
- `kilauea_obl_seismic` - Kilauea Observatory Bluff seismic (50 Hz estimated)  
- `kilauea_uwe_seismic` - Kilauea Uwekauhane seismic (100 Hz)
- `kilauea_uwe_infrasound` - Kilauea Uwekauhane infrasound (100 Hz)

### Current Limitation:
The `main()` function currently has hardcoded Mt. Spurr parameters. To test other stations, we need to modify it to accept network/station/channel parameters.

### Speed-up Analysis:
- **Mt. Spurr (50 Hz)**: 1 day → 57.6 seconds (1,500x speedup)
- **Kilauea HHZ (100 Hz)**: 1 day → 115.2 seconds (750x speedup) 
- **Kilauea HDF Infrasound (100 Hz)**: 1 day → 115.2 seconds (750x speedup)

Different native sampling rates result in different playback speeds when using the same audio sampling rate!
